experimental/Schemes/src/Resolution_structure.jl

```diff
@@ -962,6 +1150,95 @@ end
 # test for snc                                                         #
 ########################################################################
 
+# check for singularities worse than A1 (they lie at ret_ideal_sheaf)
+# and count the A1 encountered on the way (their count is total_number)
+function curve_sing_A1_or_beyond(I::AbsIdealSheaf)
+  !is_one(I) || return(I,0)
+  @assert dim(I) == 1
+  I_scheme,I_inc = sub(I)
+  I_sl = pushforward(I_inc)(ideal_sheaf_of_singular_locus(I_scheme))
+  decomp = maximal_associated_points(I_sl)    # zero-dimensional, as I describes a curve
+  init_ideal_sheaf = unit_ideal_sheaf(scheme(I_sl))
+  ret_ideal_sheaf = init_ideal_sheaf
+  total_number = 0
+  for J in decomp
+    check_val,local_number = is_A1_at_point_curve(I,J)
+    if check_val
+      total_number = total_number + local_number
+    else
+      ret_ideal_sheaf = (ret_ideal_sheaf != init_ideal_sheaf ? ret_ideal_sheaf * J : J)
+    end
+  end
+  return ret_ideal_sheaf, total_number
+end
+
+function is_A1_at_point_curve(IX::AbsIdealSheaf,Ipt::AbsIdealSheaf)
+  @assert scheme(IX) === scheme(Ipt)
+  @assert dim(scheme(IX)) == 2
+  @assert dim(Ipt) == 0
+
+  patches_scheme = patches(simplified_covering(scheme(Ipt)))
+  found_index = findfirst(V -> !is_one(Ipt(V)), patches_scheme)
+  U = patches_scheme[found_index]
+  IXsat = saturated_ideal(IX(U))
+  R = base_ring(IXsat)
+  IXU = ideal(R,small_generating_set(IXsat))
+  IptU = saturated_ideal(Ipt(U))
+
+  # absolutely irreducible point
+  if vector_space_dimension(quo(R,IptU)[1]) == 1
+    return (check_A1_at_point_curve(IXU,IptU) ? (true,1) : (false,0))
+  else
+    decomp = absolute_primary_decomposition(IptU)
+    length(decomp) == 1 || error("decomposition problem of point")
+    mult = decomp[1][4]
+    I_kbar = decomp[1][3]
+    r_changed = base_ring(I_kbar)
+    kk = coefficient_ring(r_changed)
+    IXU_changed = ideal(r_changed,  [change_coefficient_ring(kk,a, parent = r_changed) for a=gens(IXU)])
+    IptU_changed = ideal(r_changed,  [change_coefficient_ring(kk,a, parent = r_changed) for a=gens(IptU)])
+    return (check_A1_at_point_curve(IXU_changed,IptU_changed) ? (true, mult) : (false,0))
+  end
+end
+
+function check_A1_at_point_curve(IX::Ideal, Ipt::Ideal)
+## only call this from higher functions
+## it assumes: IX singular at Ipt, germ of IX at Ipt contact equivalent to hypersurface singularity
+  R = base_ring(IX)
+  dim(IX) == 1 || error("not applicable: not a curve")
+  R == base_ring(Ipt) || error("basering mismatch")
+  kk = base_ring(R)
+  characteristic(kk) == 0 || error("only available in characteristic zero")
+
+  JM = jacobian_matrix(gens(IX))
+
+  ## localize at point
+  a = rational_point_coordinates(Ipt)
+  U = complement_of_point_ideal(R,a)
+  RL, loc_map = localization(R,U)
+  IX_loc = loc_map(IX)
+  JM_loc =  map(x ->loc_map(x), JM[:,:])
+
+  if !all(iszero(a))
```

Comment:
I am confused by this. Isn't `iszero(a)` just a boolean? Then this could be changed to
```suggestion
  if !iszero(a)
```
But maybe `iszero(a)` is something else here?

Reply:
Indeed -- leftover from a different condition at this place.